lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php

```diff
@@ -95,69 +95,22 @@ public function __construct(
     }
 
     /**
-     * Add cc address
-     *
-     * @param array|string $address
-     * @param string $name
-     * @return $this
-     */
-    public function addCc($address, $name = '')
-    {
-        $this->message->addCc($address, $name);
-        return $this;
-    }
-
-    /**
-     * Add to address
-     *
-     * @param array|string $address
-     * @param string $name
-     * @return $this
-     */
-    public function addTo($address, $name = '')
-    {
-        $this->message->addTo($address, $name);
-        return $this;
-    }
-
-    /**
-     * Add bcc address
-     *
-     * @param array|string $address
-     * @return $this
+     * @return \Magento\Framework\Mail\MessageInterface
      */
-    public function addBcc($address)
+    public function getMessage()
```

Comment:
I am currently following these discussions with interest from the sideline. I am looking at ways of how to best port my EmailAttachments extension to Magento 2. In Magento 1 I used a rewrite on the Template Mailer https://github.com/fooman/emailattachments/blob/master/app/code/community/Fooman/EmailAttachments/Model/Core/Email/Template/Mailer.php#L54 to in the end allow me to add an attachment to the message. Obviously being able to do the same without a rewrite (di preference) would be great.

My current reading of the Magento2 code keeps the message object very private. So far I haven't seen a good entry point to access the message to be able to manipulate more than the template content. A getMessage() to access the message object for further manipulation seems a good starting point but maybe there is a better way which I haven't come across yet.